crates/apps/src/lib/cli.rs

```diff
@@ -6090,7 +6390,19 @@ pub mod args {
                 signatures: self
                     .signatures
                     .iter()
-                    .map(|path| std::fs::read(path).unwrap())
+                    .map(|path| {
+                        std::fs::read(path)
+                            .map_err(|e| {
+                                std::io::Error::new(
+                                    std::io::ErrorKind::InvalidInput,
+                                    format!(
+                                        "Error reading signature file: {}",
+                                        e
+                                    ),
+                                )
+                            })
+                            .unwrap()
+                    })
                     .collect(),
```

Comment:
```suggestion
                    .collect()?,
```
you can collect a `Result<C, _>` for some collection `C` from an iterator yielding `Result` values

Reply:
```rs
.collect::<Result<Vec<_>, _>>
```